nb2/bot/slack_bot.py

```diff
@@ -182,6 +215,23 @@ def is_remember_action(self, command: str) -> bool:
 
         return True
 
+    def is_remind_action(self, command: str) -> bool:
+        """
+        Return True if the command string indicates a call to NB's "remind" action.
+
+        A valid "remind" action is a command of the form:
+        '<@NB_user_id> remind (me | <@user_id_to_remind>+ ) of <@user_id_to_remember>'
+
+        Notes:
+        - <@NB_user_id> is optional; it may not be present if a command is issued
+          in a DM with the bot.
+        - Tag the sender in the response if remind is followed by "me", otherwise
+          tag  <@user_id_to_remind>
+        - Only one target user is allowed.
+        """
+        regex_pattern = "^remind\\W+(me|(<@.*>)+)\\W+of\\W+<@.*>\\W*$"
+        return re.match(regex_pattern, command, re.I) is not None
```

Comment:
While doing `is_converse_action()`, I realized that it'll basically be the same method except for `regex_pattern`. I wonder if we can just have a super generic method that just takes in the regex pattern. Then again, having this "wrapper" method is kinda nice in making the intent obvious.

But on the other hand, wouldn't it be the same thing if it was just some boolean:
`is_remember_action = re.match(pattern, command, re.I)`.

Hmm.

Reply:
Yeah, earlier on I'd been thinking it would be good to just pair the regex pattern to the function that gets called.

something like:
```python
def remind(some, params)
    # does the stuff needed for remind

def run_action(command):
    action_mapping = {
        # regex to test, function object
        "^remind\\W+(me|(<@.*>)+)\\W+of\\W+<@.*>\\W*$": remind,
        # .... more mappings
    }

    for regex_pattern, fnc in action_mapping.items():
        if re.match(regex_pattern, command, re.I) is not None:
            return fnc(params)
```

The main issue is that each function would need the same signature. But I think we could make that work.

Reply:
The funny part is, before all these action matching functions, I was honestly just thinking of just using some dict that maps the regex to the action functions like you have here.

But instead of the iteration below, couldn't it be another switch (I don't think there's switch in Python), like:
```
switch(action_mapping):
  case remind:
      remind(<unique function signature>)
  case quote:
      quote(<blah blah blah>)
...
```

Reply:
Yeah I don't think there's a switch in Python either :( I think the usual way to handle it is with a dict and then some loop or series of if statements.